src/main/java/nextstep/subway/line/application/LineService.java

```diff
@@ -12,42 +13,34 @@
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
-import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.List;
-import java.util.Optional;
 import java.util.stream.Collectors;
 
 @Service
 @Transactional
 public class LineService {
-    private LineRepository lineRepository;
-    private StationService stationService;
+    private final LineRepository lineRepository;
+    private final StationService stationService;
 
-    public LineService(LineRepository lineRepository, StationService stationService) {
+    private final SectionRepository sectionRepository;
+
+    public LineService(LineRepository lineRepository, StationService stationService, SectionRepository sectionRepository) {
         this.lineRepository = lineRepository;
         this.stationService = stationService;
+        this.sectionRepository = sectionRepository;
     }
 
     public LineResponse saveLine(LineRequest request) {
-        Station upStation = stationService.findById(request.getUpStationId());
-        Station downStation = stationService.findById(request.getDownStationId());
-        Line persistLine = lineRepository.save(new Line(request.getName(), request.getColor(), upStation, downStation, request.getDistance()));
-        List<StationResponse> stations = getStations(persistLine).stream()
-                .map(it -> StationResponse.of(it))
-                .collect(Collectors.toList());
-        return LineResponse.of(persistLine, stations);
+        final Station upStation = stationService.findById(request.getUpStationId());
+        final Station downStation = stationService.findById(request.getDownStationId());
+        final Section savedSection = sectionRepository.save(new Section(request.toLine(), upStation, downStation, request.getDistance()));
+        final Line savedLine = savedSection.getLine();
+        return LineResponse.of(savedLine, stationResponsesBy(savedLine.getStations()));
     }
 
     public List<LineResponse> findLines() {
```

Comment:
@Transactional(readOnly = true) 를 사용하면 어떠한 이점이 있을까요? 😄 
```suggestion
    @Transactional(readOnly = true)
    public List<LineResponse> findLines() {
```

Reply:
전체적으로 read 권한만 가져서 안전성 면에서 더 좋고 필요한 부분만 @transactional 만 부여해서 다른 속성으로 처리하는게 
바람직하다고 생각됩니다.
구현하기에만 집중한 나머지 다른곳을 신경쓰지 않았네요 ^^;; 
수정 하도록 하겠습니다. 감사합니다.